assets/js/editor-components/product-control/index.tsx

```diff
@@ -142,29 +165,37 @@ const ProductControl = ( {
 		} else if ( showVariations ) {
 			return renderItemWithVariations;
 		}
-		return null;
```

Comment:
This is a behaviour change, but returning null was strictly wrong, we attempt to call this as a function in underlying components. I guess its possible it never came up if the first 2 conditions are always met.